### PR TITLE
Changed host ip lookup to IPv4

### PR DESCRIPTION
**Pull request by maxhardt**

### CODE REVIEW

File: aws/main.tf [hunk 1]
```diff
@@ -125,14 +125,15 @@ resource "aws_key_pair" "cdp_keypair" {
 
 # ------- Lookup public ip for ingress settings if ingress_extra_cidrs_and_ports variable is not specified
 locals {
-  # flag to determine if public ip lookup should be performed for ingress
+  # flag to determine if public ip of host should be used for ingress
   lookup_ip = var.ingress_extra_cidrs_and_ports == null ? true : false
+  host_public_ip = "${chomp(data.http.my_ip[0].response_body)}/32"
```
Comment (jimright) [inline aws/main.tf:130]: This assignment always occurs and so will fail when `lookup_ip` is false (i.e. when the ingress_extra_cidrs_and_ports input variable is set).

Easiest way to get around this is to put `${chomp(data.http.my_ip[0].response_body)}/32` expression on L136.

File: azure/main.tf [hunk 2]
```diff
@@ -113,14 +113,15 @@ resource "local_sensitive_file" "pem_file" {
 
 # ------- Lookup public ip for ingress settings if ingress_extra_cidrs_and_ports variable is not specified
 locals {
-  # flag to determine if public ip lookup should be performed for ingress
+  # flag to determine if public ip of host should be used for ingress
   lookup_ip = var.ingress_extra_cidrs_and_ports == null ? true : false
+  host_public_ip = "${chomp(data.http.my_ip[0].response_body)}/32"
```
Comment (jimright) [inline azure/main.tf:118]: Same as for AWS, this fails when `lookup_ip` is false.

File: gcp/main.tf [hunk 3]
```diff
@@ -100,14 +100,15 @@ resource "local_sensitive_file" "pem_file" {
 
 # ------- Lookup public ip for ingress settings if ingress_extra_cidrs_and_ports variable is not specified
 locals {
-  # flag to determine if public ip lookup should be performed for ingress
+  # flag to determine if public ip of host should be used for ingress
   lookup_ip = var.ingress_extra_cidrs_and_ports == null ? true : false
+  host_public_ip = "${chomp(data.http.my_ip[0].response_body)}/32"
```
Comment (jimright) [inline gcp/main.tf:105]: Same as for AWS, this fails when `lookup_ip` is false.

**Review by jimright — CHANGES_REQUESTED** [review]
Hi @maxhardt,

Thanks for handling this. See comments for one set of changes required to handle the case where the `ingress_extra_cidrs_and_ports` input variable is set.

Also it looks like your commit is missing DCO signoff (you had this on https://github.com/cloudera-labs/terraform-cdp-modules/pull/53 so might just be a git config setting).

**Comment by maxhardt** [comment]
Thanks for catching these @jimright, followed your suggestion in the latest commits. Would be great to have some rudimentary (unit) testing to catch these when developing as well 🧪

**Comment by jimright** [comment]
> Thanks for catching these @jimright, followed your suggestion in the latest commits. Would be great to have some rudimentary (unit) testing to catch these when developing as well, happy to contribute! 🧪

Yes, the addition of some testing is great suggestion and would definitely help to catch potential issues or regressions early.
We've done some terratest for another project but Terraform have also recently released a native `terraform test` framework that should help. Here's an overview - https://www.hashicorp.com/blog/testing-hashicorp-terraform. Let's review this and see what we can achieve.

**Review by jimright — APPROVED** [review]
Looks good. Approved.

**Pull request merged**